common/src/main/scala/uk/ac/wellcome/platform/stacks/common/services/AkkaClientService.scala

```diff
@@ -0,0 +1,100 @@
+package uk.ac.wellcome.platform.stacks.common.services
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.marshalling.{Marshal, Marshaller}
+import akka.http.scaladsl.model.Uri.Query
+import akka.http.scaladsl.model.headers.{Authorization, BasicHttpCredentials, OAuth2BearerToken}
+import akka.http.scaladsl.model.{HttpEntity, HttpHeader, HttpMethods, HttpRequest, HttpResponse, RequestEntity, Uri}
+import akka.http.scaladsl.unmarshalling.{Unmarshal, Unmarshaller}
+import akka.stream.ActorMaterializer
+
+import scala.concurrent.{ExecutionContextExecutor, Future}
+
+trait AkkaClientService {
+  implicit val system: ActorSystem
+  implicit val mat: ActorMaterializer
+
+  implicit val ec: ExecutionContextExecutor = system.dispatcher
+
+  protected val baseUri: Uri
+
+  protected def buildUri(path: String, params: Map[String, String] = Map.empty): Uri =
+    baseUri
+      .copy(path = baseUri.path + "/" + path)
```

Comment:
Would it be worth using `java.nio.file.Paths` and `Paths.get` to join these two, like we do for ObjectLocation? https://github.com/wellcometrust/scala-storage/blob/master/storage/src/main/scala/uk/ac/wellcome/storage/ObjectLocation.scala

Reply:
There is an Akka Path object that may well suit too.